sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/arm/ResourceUtils.java

```diff
@@ -211,4 +211,17 @@ public static String constructResourceId(
                 resourceType,
                 resourceName);
     }
+
+    /**
+     * Properly encode resource ID in path parameter.
+     *
+     * @param resourceId the resource ID to encode
+     * @return properly encoded resource ID
+     */
+    public static String encodeResourceId(String resourceId) {
```

Comment:
We should at least mention in javadoc, that the API is to ensure resourceId is properly encoded (instead of just encode it), as we actually don't know whether the resourceId we get from backend is proper encoded or not.

(same to the `boolean encodeResourceId` above on monitor)

Reply:
Added javadoc.